spec/src/main/asciidoc/core/definition.asciidoc

```diff
@@ -493,13 +493,14 @@ Alternatively, a scope type may be specified using a stereotype annotation, as d
 
 ==== Default scope
 
-When no scope is explicitly declared by annotating the bean class or producer method or field the scope of a bean is defaulted.
+When no scope is explicitly declared by annotating the bean class or producer method or field and the bean is not deployed in a <<trimmed_bean_archive>> the scope of a bean is defaulted.
```

Comment:
cannot follow you. 'bean class' in that case most likely doesn't refer to Bean<T>#getBeanClass() but 'managed bean'.